lib/_emerge/depgraph.py

```diff
@@ -2612,18 +2625,21 @@ def _reinstall_for_flags(self, pkg, forced_flags,
 			_get_eapi_attrs(pkg.eapi))
 
 		if newuse or (binpkg_respect_use and not changed_use):
-			flags = set(orig_iuse.symmetric_difference(
-				cur_iuse).difference(forced_flags))
-			flags.update(orig_iuse.intersection(orig_use).symmetric_difference(
-				cur_iuse.intersection(cur_use)))
-			flags.difference_update(feature_flags)
+			flags = set(orig_iuse ^ cur_iuse) - forced_flags
+			flags |= (
+				orig_iuse.intersection(orig_use)
+				^ cur_iuse.intersection(cur_use)
+			)
+			flags -= feature_flags
 			if flags:
 				return flags
 
 		elif changed_use or binpkg_respect_use:
-			flags = set(orig_iuse.intersection(orig_use).symmetric_difference(
-				cur_iuse.intersection(cur_use)))
-			flags.difference_update(feature_flags)
+			flags = set(
```

Comment:
Redundant `set(` here.

Reply:
It isn't redundant as orig_iuse is sometimes a `frozenset`, and a new `frozenset` wouldn't be compatible with the downstream operations performed on `flags`.

Reply:
Per discussion w/ @zmedico in IRC, will collapse the later update into this arithmetic so that a `frozenset` is acceptable

Reply:
Adjusted this change to make an initial copy to a `set` followed by in-place mutations

Reply:
Adjusted this change to make an initial copy to a set followed by in-place mutations.